registrations/tests.py

```diff
@@ -0,0 +1,299 @@
+
+    def test_login_normaluser(self):
+        """ Test that normaluser can login successfully
+        """
+        # Setup
+        post_auth = {"username": "testnormaluser",
+                     "password": "testnormalpass"}
+        # Execute
+        request = self.client.post(
+            '/api/token-auth/', post_auth)
+        token = request.data.get('token', None)
+        # Check
+        self.assertIsNotNone(
+            token, "Could not receive authentication token on login post.")
+        self.assertEqual(
+            request.status_code, 200,
+            "Status code on /api/token-auth was %s (should be 200)."
+            % request.status_code)
+
+    def test_login_adminuser(self):
+        """ Test that adminuser can login successfully
+        """
+        # Setup
+        post_auth = {"username": "testadminuser",
+                     "password": "testadminpass"}
+        # Execute
+        request = self.client.post(
+            '/api/token-auth/', post_auth)
+        token = request.data.get('token', None)
+        # Check
+        self.assertIsNotNone(
+            token, "Could not receive authentication token on login post.")
+        self.assertEqual(
+            request.status_code, 200,
+            "Status code on /api/token-auth was %s (should be 200)."
+            % request.status_code)
+
+    def test_login_adminuser_wrong_password(self):
+        """ Test that adminuser cannot log in with wrong password
+        """
+        # Setup
+        post_auth = {"username": "testadminuser",
+                     "password": "wrongpass"}
+        # Execute
+        request = self.client.post(
+            '/api/token-auth/', post_auth)
+        token = request.data.get('token', None)
+        # Check
+        self.assertIsNone(
+            token, "Could not receive authentication token on login post.")
+        self.assertEqual(request.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_login_otheruser(self):
+        """ Test that an unknown user cannot log in
+        """
+        # Setup
+        post_auth = {"username": "testotheruser",
+                     "password": "testotherpass"}
+        # Execute
+        request = self.otherclient.post(
+            '/api/token-auth/', post_auth)
+        token = request.data.get('token', None)
+        # Check
+        self.assertIsNone(
+            token, "Could not receive authentication token on login post.")
+        self.assertEqual(request.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class TestSourceAPI(AuthenticatedAPITestCase):
+
+    def test_get_source_adminuser(self):
+        # Setup
+        source = self.make_source_adminuser()
+        # Execute
+        response = self.adminclient.get('/api/v1/source/%s/' % source.id,
+                                        format='json',
+                                        content_type='application/json')
+        # Check
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Please can we check some of the `response.data["field"]` values are correctly serialized